packages/@lwc/integration-karma/test/component/lifecycle-callbacks/invocationorder/lightFooInternal/lightFooInternal.js

```diff
@@ -0,0 +1,19 @@
+import { LightningElement } from 'lwc';
+
+export default class FooInternal extends LightningElement {
+    static renderMode = 'light';
+    get externalClassName() {
+        return this.getAttribute('class');
+    }
+    connectedCallback() {
+        window.timingBuffer.push(`${this.externalClassName}:connectedCallback`);
+    }
+    renderedCallback() {
+        window.timingBuffer.push(`${this.externalClassName}:renderedCallback`);
+    }
+    disconnectedCallback() {
+        if (window.timingBuffer) {
```

Comment:
Curious why we need this check for the disconnected case.

Reply:
I'll add a comment. It's because our Karma tests remove the components after the test completes, which means it will trigger an error since the `afterEach` callbacks occur in such an order that `window.timingBuffer` has already been deleted.